packages/serverpod/lib/src/cache/redis_cache.dart

```diff
@@ -45,11 +46,11 @@ class RedisCache extends GlobalCache {
       'Redis needs to be enabled to use this method',
     );
     var data = await redisController!.get(key);
-    if (data == null) {
+    if (data.isNullOrEmpty) {
       return null;
     }
 
-    return serializationManager.decode<T>(data, t);
+    return serializationManager.decode<T>(data!, t);
```

Comment:
Note: We want to avoid using `!` to dereference possible nullable values and are actively working on removing them from the code. This is another issues that is surfaced when grouping the meaning of null and empty. We lose the Darts native null check support.

Reply:
But as of the code we have [here](https://github.com/serverpod/serverpod/blob/0b47da20ebb6ceb47bfa0eb8b6f8089d120df222/packages/serverpod_serialization/lib/src/serialization.dart#L39), we better pass no null string.

Reply:
I think we misunderstood each other here. You are correct that we can't and shouldn't pass null to the serializationManager.

But with the old code `data == null` the dart analyzer can deduce that data is never null at this point and therefore we don't need the scary `!`. But using the method introduced the analyzer can't make does deductions and we loose that support.

That is one of many reasons we should not introduce an `isNull` method.

Therefore I would like to see this method remove and all uses of it in this PR removed as well.